dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs

```diff
@@ -24,25 +24,25 @@ public sealed class SequentialPlanner : ISequentialPlanner
     /// </summary>
     /// <param name="kernel">The semantic kernel instance.</param>
     /// <param name="config">The planner configuration.</param>
-    /// <param name="prompt">Optional prompt override</param>
     public SequentialPlanner(
         IKernel kernel,
-        SequentialPlannerConfig? config = null,
-        string? prompt = null)
+        SequentialPlannerConfig? config = null)
     {
         Verify.NotNull(kernel);
-        this.Config = config ?? new();
 
-        this.Config.ExcludedSkills.Add(RestrictedSkillName);
+        // Set up config with default value and excluded skills
+        this._config = config ?? new();
+        this._config.ExcludedSkills.Add(RestrictedSkillName);
 
-        string promptTemplate = prompt ?? EmbeddedResource.Read("skprompt.txt");
+        // Set up prompt template
+        string promptTemplate = this._config.GetPromptTemplate?.Invoke() ?? EmbeddedResource.Read("skprompt.txt");
 
         this._functionFlowFunction = kernel.CreateSemanticFunction(
             promptTemplate: promptTemplate,
             skillName: RestrictedSkillName,
             description: "Given a request or command or goal generate a step by step plan to " +
                          "fulfill the request using functions. This ability is also known as decision making and function flow",
-            maxTokens: this.Config.MaxTokens ?? 1024,
+            maxTokens: this._config.MaxTokens ?? 1024,
```

Comment:
This stuff is going to be changing with the `AIRequestSettings` work. Just a heads up.